File: sdks/sdkwork-mail-sdk/sdkwork-mail-sdk-swift/Sources/MailSdk/mailDriverManager.swift
public struct MailDriverManager {
    public init() {}

    public func resolveSelection(
        request: MailProviderSelectionRequest = MailProviderSelectionRequest(),
        defaultProviderKey: String = MailProviderCatalog.DEFAULT_mail_PROVIDER_KEY
    ) -> MailProviderSelection {
        return MailProviderSelection.resolveMailProviderSelection(
            request: request,
            defaultProviderKey: defaultProviderKey
        )
    }

    public func describeProviderSupport(providerKey: String) -> MailProviderSupport {
        let official = MailProviderCatalog.getMailProviderByProviderKey(providerKey) != nil
        let activation = MailProviderActivationCatalog.getMailProviderActivationByProviderKey(providerKey)

        return MailProviderSupport.createMailProviderSupportState(
            .init(
                providerKey: providerKey,
                builtin: activation?.builtin ?? false,
                official: official,
                registered: activation?.runtimeBridge ?? false
            )
        )
    }

    public func listProviderSupport() -> [MailProviderSupport] {
        MailProviderCatalog.entries.map { describeProviderSupport(providerKey: $0.providerKey) }
    }
}

File: sdks/sdkwork-mail-sdk/sdkwork-mail-sdk-csharp/src/SDKWork.Mail.Sdk/mailDriverManager.cs
namespace Sdkwork.Mail.Sdk;

using System.Linq;

public sealed class MailDriverManager
{
    public MailProviderSelection ResolveSelection(
        MailProviderSelectionRequest? request = null,
        string defaultProviderKey = MailProviderCatalog.DEFAULT_mail_PROVIDER_KEY
    )
    {
        return MailProviderSelection.ResolveMailProviderSelection(request, defaultProviderKey);
    }

    public MailProviderSupport DescribeProviderSupport(string providerKey)
    {
        var official = MailProviderCatalog.GetMailProviderByProviderKey(providerKey) is not null;
        var activation = MailProviderActivationCatalog.GetMailProviderActivationByProviderKey(providerKey);

        return MailProviderSupport.CreateMailProviderSupportState(
            new MailProviderSupportStateRequest(
                providerKey,
                activation?.builtin ?? false,
                official,
                activation?.runtimeBridge ?? false
            )
        );
    }

    public IReadOnlyList<MailProviderSupport> ListProviderSupport()
    {
        return MailProviderCatalog.Entries
            .Select(entry => DescribeProviderSupport(entry.providerKey))
            .ToArray();
    }
}
